sklearn/experimental/_column_transformer.py

```diff
@@ -0,0 +1,282 @@
+"""
+The :mod:`sklearn.experimental._column_transformer` module implements utilities
+to work with heterogeneous data and to apply different transformers to
+different columns.
+"""
+# Author: Andreas Mueller
+#         Joris Van den Bossche
+# License: BSD
+
+
+import numpy as np
+
+from ..externals import six
+from ..pipeline import FeatureUnion, _name_estimators
+
+__all__ = ['ColumnTransformer', 'make_column_transformer']
+
+
+_ERR_MSG_1DCOLUMN = ("1D data passed to a transformer that expects 2D data. "
+                     "Try to specify the column selection as a list of one "
+                     "item instead of a scalar.")
+
+
+class ColumnTransformer(FeatureUnion):
+    """Applies transformers to columns of an array or pandas DataFrame.
+
+    This estimator applies transformer objects to columns of the input,
+    then concatenates the results. This is useful for heterogeneous or
+    columnar data, to combine several feature extraction mechanisms into a
+    single transformer.
+
+    Read more in the :ref:`User Guide <column_transformer>`.
+
+    Parameters
+    ----------
+    transformer_list : list of tuples
+        List of (name, transformer, column) tuples specifying the transformer
+        objects to be applied to subsets of the data. The columns can be
+        specified as a scalar or slice/list (for multiple columns) of integer
+        or string values, or a boolean mask. Integers are interpreted as the
+        positional columns, strings as the column labels of `X`.
+        When passing a single column to a transformer that expects 2D input
+        data, the column should be specified a list of one element.
```

Comment:
We need to clarify somewhere the behaviour when a DataFrame is sliced: will the transformers receive Pandas types (DataFrame, Series), or numpy/scipy types?